R/zchunk_LA143.HDDCDD-gcam-usa.R

```diff
@@ -33,91 +35,319 @@ module_gcam.usa_LA143.HDDCDD_DISABLED <- function(command, ...) {
+      Census_pop_hist_subregion
+
+    # Add subregion13 (for RECS) and subregion9 (for CBECS) columns to the historic cooling degree days
+    # latter and heating degree days. Latter on this will be used to  to calculate the share of person heating
+    # degree days within census division (subregion).
     #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
+    # Historic cooling degree days.
+    CDD_His %>%
+      left_join_error_no_match(states_subregions %>%
+                                 select(state, subregion9, subregion13),
+                               by = "state") %>%
+      gather_years ->
+      CDD_His_subregion
+
+    # Historic heating degree days.
+    HDD_His %>%
+      left_join_error_no_match(states_subregions %>%
+                                 select(state, subregion9, subregion13),
+                               by = "state") %>%
+      gather_years ->
+      HDD_His_subregion
+
+
+    # Interpolate to fill out any missing historical years cooling degree days.
+    CDD_His_subregion %>%
+      complete(nesting(state, subregion9, subregion13), year = c(year, HISTORICAL_YEARS)) %>%
+      arrange(state, subregion9, subregion13, year) %>%
+      group_by(state, subregion9, subregion13) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup ->
+      L143.CDD_state
+
+    # Interpolate to fill out any missing historical years heating degree days.
+    HDD_His_subregion %>%
+      complete(nesting(state, subregion9, subregion13), year = c(year, HISTORICAL_YEARS)) %>%
+      arrange(state, subregion9, subregion13, year) %>%
+      group_by(state, subregion9, subregion13) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup ->
+      L143.HDD_state
+
+
+    # Multiply the historic cooling degree days tibble by the census population to calculate
+    # the population cooling degree days.
+    L143.CDD_state %>%
+      rename(degree_day = value) %>%
+      left_join_error_no_match(Census_pop_hist_subregion %>% rename(population = value),
+                               by = c("year", "state", "subregion9", "subregion13")) %>%
+      mutate(value = degree_day * population) %>%
+      select(state, subregion9, subregion13, year, value) ->
+      L143.Pop_CDD_state
+
+    # Multiply the historic heating degree days tibble by the census population to calculate
+    # the population heating degree days.
+    L143.HDD_state %>%
+      rename(degree_day = value) %>%
+      left_join_error_no_match(Census_pop_hist_subregion %>% rename(population = value),
+                               by = c("year", "state", "subregion9", "subregion13")) %>%
+      mutate(value = degree_day * population) %>%
+      select(state, subregion9, subregion13, year, value) ->
+      L143.Pop_HDD_state
+
+    # Aggregate the population cooling and heating degree days by census subregions.
```

Comment:
Unclear to me why the aggregation for each subregion needs to be done one by one.

Reply:
It would be  helful to add a comment on why this aggregation needs to be done.